verafelix/reprograma.py

```diff
@@ -0,0 +1,202 @@
+        "Participação": 7.2
+    },
+    ("Camila", "Pereira"): {
+        "Turma": "Turma B",
+        "Notas": [7.5, 8.0, 8.5],
+        "Presença": [True, True, True, True, True],
+        "Participação": 8.0
+    },
+    ("Luana", "Rodrigues"): {
+        "Turma": "Turma A",
+        "Notas": [9.0, 9.0, 8.5],
+        "Presença": [True, True, True, True, True],
+        "Participação": 8.8
+    },
+    ("Beatriz", "Lima"): {
+        "Turma": "Turma C",
+        "Notas": [8.0, 7.5, 7.0],
+        "Presença": [True, True, True, False, True],
+        "Participação": 7.5
+    }
+}
+
+
+def main():
+    print("\n---  Seja bem vinda a Escola do Reprograma!  ---")
+    print("Sistema de informações de alunas")
+
+    while True:
+        cod_opcao = obter_opcao()  # Obtém a opção escolhida pelo usuário
+
+        if cod_opcao == 1:
+            incluir_nova_aluna()  # Chama a função para incluir nova aluna
+        elif cod_opcao == 2:
+            consultar_lista_alunas()  # Chama a função para consultar lista de alunas
+        elif cod_opcao == 3:
+            consultar_faltas_aluna()  # Chama a função para consultar faltas de uma aluna
+        elif cod_opcao == 4:
+            consultar_notas_aluna()  # Chama a função para consultar notas de uma aluna
+        elif cod_opcao == 5:
+            consultar_status_aprovacao()  # Chama a função para consultar status de aprovação
+        elif cod_opcao == 6:
+            print("Encerrando o programa...")
+            break  # Encerra o programa
+
+
+def obter_opcao():
+    codigo_opcao = 0
+
+    while codigo_opcao not in [1, 2, 3, 4, 5, 6]:
+        try:
+            # Solicita ao usuário para escolher uma opção
+            codigo_opcao = int(input("\nEscolha uma opção:\n"
+                                     "1 - Incluir uma nova aluna\n"
+                                     "2 - Consultar lista de alunas\n"
+                                     "3 - Consultar faltas da aluna\n"
+                                     "4 - Consultar notas da aluna\n"
+                                     "5 - Consultar status de aprovação\n"
+                                     "6 - Sair do sistema\n"
+                                     "Opção: "))
+
+            if codigo_opcao not in [1, 2, 3, 4, 5, 6]:
+                print("Opção inválida. Por favor, escolha uma opção válida (1 a 6).\n")
+        except ValueError:
+            print("Entrada inválida. Por favor, digite um número inteiro.\n")
+
+    return codigo_opcao
+
+
+def incluir_nova_aluna():
+    # Solicita os dados da nova aluna
+    nome = input("Digite o nome da aluna: ")
+    sobrenome = input("Digite o sobrenome da aluna: ")
+    turma = input("Digite a turma da aluna: ")
+    notas = list(
+        map(float, input("Digite as notas da aluna separadas por espaço: ").split()))
+    presenca = list(map(lambda x: x.lower() == 'true', input(
+        "Digite as presenças da aluna (True/False) separadas por espaço: ").split()))
```

Comment:
Bom uso do map no Python, porém, deixar tantas funções encadeadas em uma única linha pode dificultar a legibilidade do código. Cuidado no modo utiliza referências externas para ajudar, pode acabar atrapalhando no processo de aprendizado neste início.

**Sugestão:**
```suggestion
    notas = input("Digite as notas da aluna (separadas por espaço): ")
    notas = list(map(float, notas.split()))

    presencas = input("Digite a presença da aluna (True/False separados por espaço): ")
    presencas = list(map(lambda x: x.lower() == 'true', presencas.split()))
```